Mathlib/Algebra/Lie/Classical.lean

```diff
@@ -96,30 +96,74 @@ theorem sl_bracket [Fintype n] (A B : sl n R) : ⁅A, B⁆.val = A.val * B.val -
 
 section ElementaryBasis
 
-variable {n R} [Fintype n] (i j : n)
+variable {n R} [Fintype n] (i j k : n)
 
-/-- When `i ≠ j`, the elementary matrices are elements of `sl n R`, in fact they are part of a
-natural basis of `sl n R`. -/
-def Eb (h : i ≠ j) (r : R) : sl n R :=
-  ⟨Matrix.single i j r,
-    show Matrix.single i j r ∈ LinearMap.ker (Matrix.traceLinearMap n R R) from
-      Matrix.trace_single_eq_of_ne i j r h⟩
+/-- When `i ≠ j`, the single-element matrices are elements of `sl n R`.
+
+Along with some elements produced by `singleSubSingle`, these form a natural basis of `sl n R`. -/
+def single (h : i ≠ j) (r : R) : sl n R :=
+  ⟨Matrix.single i j r, Matrix.trace_single_eq_of_ne i j r h⟩
+
+@[deprecated (since := "2025-05-06")] alias Eb := single
+
+@[simp]
+theorem val_single (h : i ≠ j) (r : R) : (single i j h r).val = Matrix.single i j r :=
+  rfl
+
+@[simp]
+theorem single_zero (h : i ≠ j) : single i j h (0 : R) = 0 := Subtype.ext <| Matrix.single_zero _ _
+
+@[simp]
+theorem single_add (h : i ≠ j) (r₁ r₂ : R) :
+    single i j h (r₁ + r₂) = single i j h r₁ + single i j h r₂ :=
+  Subtype.ext <| Matrix.single_add _ _ _ _
```

Comment:
How about we bundle this up as a linear map?
```suggestion
def single (h : i ≠ j) : R →ₗ[R] sl n R where
  toFun r := ⟨Matrix.single i j r, Matrix.trace_single_eq_of_ne i j r h⟩
  map_add' _ _ := Subtype.ext <| Matrix.single_add _ _ _ _
  map_smul' _ _ := Subtype.ext <| by simp
```
(and if so, likewise for `singleSubSingle`)

Reply:
Done